elliptic-curve/src/point/non_identity.rs

```diff
@@ -72,6 +72,16 @@ where
             point: self.point.to_affine(),
         }
     }
+
+    /// Multiply by the generator of the prime-order subgroup.
+    pub fn mul_by_generator<C: CurveArithmetic>(scalar: NonZeroScalar<C>) -> Self
+    where
+        P: Copy + Mul<Scalar<C>, Output = P>,
+    {
+        Self {
+            point: P::generator() * *scalar.as_ref(),
+        }
+    }
```

Comment:
This should probably delegate to the `Group`-specific implementation (which can use precomputed lookup tables), e.g.

```suggestion
        P: Group,
    {
        Self {
            point: P::mul_by_generator(scalar),
        }
    }
```

Reply:
Nice! Done!